crates/opactl/src/main.rs

```diff
@@ -82,7 +91,9 @@ async fn ambient_transactions<
         debug!(waiting_for=?goal_tx_id, max_steps=?max_steps);
         let goal_clone = goal_tx_id.clone();
 
-        let stream = reader.state_updates(None, Some(max_steps)).await;
+        let mut stream = reader
```

Comment:
You're getting an unnecessary_mut clippy warning on this definition.
```suggestion
        let stream = reader
```